OCaml 5.6 changes the float-conversion primitives in Nativeint, Int64,
and Int32. Match their names only on compilers that provide them.

--- a/bundles/stdcompat/stdcompat-current/configure.ac
+++ b/bundles/stdcompat/stdcompat-current/configure.ac
@@ -187,6 +187,7 @@
 AC_WITH_OCAML_VERSION([5_3_0], [5.3.0])
 AC_WITH_OCAML_VERSION([5_4_0], [5.4.0])
 AC_WITH_OCAML_VERSION([5_5_0], [5.5.0])
+AC_WITH_OCAML_VERSION([5_6_0], [5.6.0])
 
 AS_IF([test "x$target_os_type" = "xWin32"],
   [AC_WITH_BLOCK([WIN32])],
--- a/bundles/stdcompat/stdcompat-current/stdcompat__nativeint_s.mli.in
+++ b/bundles/stdcompat/stdcompat-current/stdcompat__nativeint_s.mli.in
@@ -106,9 +106,14 @@
 (** Alias for {!Nativeint.to_int} *)
 
 @BEGIN_FROM_4_03_0@
+@BEGIN_FROM_5_6_0@
+external of_float : float -> nativeint = "%nativeint_of_float"
+@END_FROM_5_6_0@
+@BEGIN_BEFORE_5_6_0@
 external of_float :
   float -> nativeint = "caml_nativeint_of_float"
     "caml_nativeint_of_float_unboxed"[@@unboxed ][@@noalloc ]
+@END_BEFORE_5_6_0@
 @END_FROM_4_03_0@
 @BEGIN_BEFORE_4_03_0@
 @BEGIN_FROM_3_08_0@
@@ -129,9 +134,14 @@
  *)
 
 @BEGIN_FROM_4_03_0@
+@BEGIN_FROM_5_6_0@
+external to_float : nativeint -> float = "%nativeint_to_float"
+@END_FROM_5_6_0@
+@BEGIN_BEFORE_5_6_0@
 external to_float :
   nativeint -> float = "caml_nativeint_to_float"
     "caml_nativeint_to_float_unboxed"[@@unboxed ][@@noalloc ]
+@END_BEFORE_5_6_0@
 @END_FROM_4_03_0@
 @BEGIN_BEFORE_4_03_0@
 @BEGIN_FROM_3_08_0@
--- a/bundles/stdcompat/stdcompat-current/stdcompat__int64_s.mli.in
+++ b/bundles/stdcompat/stdcompat-current/stdcompat__int64_s.mli.in
@@ -102,9 +102,14 @@
 (** Alias for {!Int64.to_int} *)
 
 @BEGIN_FROM_4_03_0@
+@BEGIN_FROM_5_6_0@
+external of_float : float -> int64 = "%int64_of_float"
+@END_FROM_5_6_0@
+@BEGIN_BEFORE_5_6_0@
 external of_float :
   float -> int64 = "caml_int64_of_float" "caml_int64_of_float_unboxed"
 [@@unboxed ][@@noalloc ]
+@END_BEFORE_5_6_0@
 @END_FROM_4_03_0@
 @BEGIN_BEFORE_4_03_0@
 @BEGIN_FROM_3_08_0@
@@ -124,9 +129,14 @@
  *)
 
 @BEGIN_FROM_4_03_0@
+@BEGIN_FROM_5_6_0@
+external to_float : int64 -> float = "%int64_to_float"
+@END_FROM_5_6_0@
+@BEGIN_BEFORE_5_6_0@
 external to_float :
   int64 -> float = "caml_int64_to_float" "caml_int64_to_float_unboxed"
 [@@unboxed ][@@noalloc ]
+@END_BEFORE_5_6_0@
 @END_FROM_4_03_0@
 @BEGIN_BEFORE_4_03_0@
 @BEGIN_FROM_3_08_0@
--- a/bundles/stdcompat/stdcompat-current/stdcompat__int32_s.mli.in
+++ b/bundles/stdcompat/stdcompat-current/stdcompat__int32_s.mli.in
@@ -168,9 +168,14 @@
  *)
 
 @BEGIN_FROM_4_03_0@
+@BEGIN_FROM_5_6_0@
+external to_float : int32 -> float = "%int32_to_float"
+@END_FROM_5_6_0@
+@BEGIN_BEFORE_5_6_0@
 external to_float :
   int32 -> float = "caml_int32_to_float" "caml_int32_to_float_unboxed"
 [@@unboxed ][@@noalloc ]
+@END_BEFORE_5_6_0@
 @END_FROM_4_03_0@
 @BEGIN_BEFORE_4_03_0@
 @BEGIN_FROM_3_08_0@
